Utilities/XrdAdaptor/src/XrdStatistics.cc

```diff
@@ -0,0 +1,177 @@
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/MessageLogger/interface/JobReport.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/ServiceRegistry/interface/ActivityRegistry.h"
+
+#include "XrdRequest.h"
+#include "XrdStatistics.h"
+
+#include <chrono>
+
+using namespace XrdAdaptor;
+
+
+std::atomic<XrdSiteStatisticsInformation*> XrdSiteStatisticsInformation::m_instance;
+std::mutex XrdSiteStatisticsInformation::m_mutex;
+
+
+XrdStatisticsService::XrdStatisticsService(const edm::ParameterSet &iPS, edm::ActivityRegistry &iRegistry)
+{
+    if (iPS.getUntrackedParameter<bool>("reportToFJR", false))
+    {
+        XrdSiteStatisticsInformation::createInstance();
+    }
+
+    iRegistry.watchPostEndJob(this, &XrdStatisticsService::postEndJob);
+}
+
+
+void XrdStatisticsService::postEndJob()
+{
+    edm::Service<edm::JobReport> reportSvc;
+    if (!reportSvc.isAvailable()) {return;}
+
+    XrdSiteStatisticsInformation *instance = XrdSiteStatisticsInformation::getInstance();
+    if (!instance) {return;}
+
+    for (std::shared_ptr<XrdSiteStatistics> const &stats : instance->m_sites)
+    {
+        stats->recomputeProperties();
+        reportSvc->reportPerformanceForModule(stats->site(), "XrdSiteStatistics", stats->fjrProperties());
+    }
+}
+
+
+std::shared_ptr<XrdSiteStatistics>
+XrdSiteStatisticsInformation::getStatisticsForSite(std::string const &site)
+{
+    for (std::shared_ptr<XrdSiteStatistics> &stats : m_sites)
+    {
+        if (stats->site() == site) {return stats;}
+    }
+    m_sites.emplace_back(new XrdSiteStatistics(site));
+    return m_sites.back();
+}
+
+
+void
+XrdSiteStatisticsInformation::createInstance()
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
```

Comment:
I don't believe there is a need for the ```m_mutex``` since making multiple copies of ```XrdSiteStaticistsInformation``` doesn't appear to be a problem. So I suggest that the code be
```
if(not m_instance) {
   std::unique_ptr<XrdSiteStatisticsInformation> tmp { new XrdSiteStatisticsInformation() };
   XrdSiteStatisticsInformation* expected = nullptr;
   if( m_instance.compare_exchange_strong(expected,tmp.get()) {
      //this thread succeeded in the exchange so we need to no longer own the memory
      tmp.release();
   }
}
```